// cypress-cucumber-steps/lib/actions/click.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.When_I_click_on_label = exports.When_I_click_on_text = exports.When_I_click_on_link = exports.When_I_click_on_button = exports.When_I_click_x_y_coordinates = exports.When_I_click_position = exports.When_I_click = void 0;
var cypress_cucumber_preprocessor_1 = require("@badeball/cypress-cucumber-preprocessor");
var queries_1 = require("../queries");
var utils_1 = require("../utils");
/**
 * When I click:
 *
 * ```gherkin
 * When I click
 * ```
 *
 * Alternative:
 *
 * - {@link When_I_click_on_text | When I click on text}
 * - {@link When_I_click_position | When I click position}
 * - {@link When_I_click_x_y_coordinates | When I click x-y coordinates}
 *
 * @example
 *
 * ```gherkin
 * When I click
 * ```
 *
 * With [options](https://docs.cypress.io/api/commands/click#Arguments):
 *
 * ```gherkin
 * When I click
 *   | altKey | false |
 *   | animationDistanceThreshold | 5 |
 *   | ctrlKey | false |
 *   | log | true |
 *   | force | false |
 *   | metaKey | false |
 *   | multiple | false |
 *   | scrollBehavior | top |
 *   | shiftKey | false |
 *   | timeout | 4000 |
 *   | waitForAnimations | true |
 * ```
 *
 * @remarks
 *
 * A preceding step like {@link When_I_find_element_by_text | "When I find element by text"} is required. For example:
 *
 * ```gherkin
 * When I find element by text "Text"
 *   And I click
 * ```
 *
 * @see
 *
 * - {@link When_I_double_click | When I double-click}
 * - {@link When_I_right_click | When I right-click}
 */
function When_I_click(options) {
    (0, utils_1.getCypressElement)().click((0, utils_1.getOptions)(options));
}
exports.When_I_click = When_I_click;
(0, cypress_cucumber_preprocessor_1.When)('I click', When_I_click);
/**
 * When I click position:
 *
 * ```gherkin
 * When I click {string}
 * ```
 *
 * You can click on 9 specific positions of an element:
 *
 * ```
 *  -------------------------------------
 * | top-left        top       top-right |
 * |                                     |
 * |                                     |
 * |                                     |
 * | left          center          right |
 * |                                     |
 * |                                     |
 * |                                     |
 * | bottom-left   bottom   bottom-right |
 *  -------------------------------------
 * ```
 *
 * @example
 *
 * ```gherkin
 * When I click "top-left"
 * ```
 *
 * With [options](https://docs.cypress.io/api/commands/click#Arguments):
 *
 * ```gherkin
 * When I click "top"
 *   | altKey | false |
 *   | animationDistanceThreshold | 5 |
 *   | ctrlKey | false |
 *   | log | true |
 *   | force | false |
 *   | metaKey | false |
 *   | multiple | false |
 *   | scrollBehavior | top |
 *   | shiftKey | false |
 *   | timeout | 4000 |
 *   | waitForAnimations | true |
 * ```
 *
 * @remarks
 *
 * A preceding step like {@link When_I_find_element_by_text | "When I find element by text"} is required. For example:
 *
 * ```gherkin
 * When I find element by text "Text"
 *   And I click "top-right"
 * ```
 *
 * @see
 *
 * - {@link When_I_click | When I click}
 * - {@link When_I_click_x_y_coordinates | When I click x-y coordinates}
 */
function When_I_click_position(position, options) {
    (0, utils_1.getCypressElement)().click((0, utils_1.camelCase)(position), (0, utils_1.getOptions)(options));
}
exports.When_I_click_position = When_I_click_position;
(0, cypress_cucumber_preprocessor_1.When)('I click {string}', When_I_click_position);
/**
 * When I click x-y coordinates:
 *
 * ```gherkin
 * When I click {int}px and {int}px
 * ```
 *
 * @example
 *
 * ```gherkin
 * When I click 80px and 75px
 * ```
 *
 * With [options](https://docs.cypress.io/api/commands/click#Arguments):
 *
 * ```gherkin
 * When I click 80px and 75px
 *   | altKey | false |
 *   | animationDistanceThreshold | 5 |
 *   | ctrlKey | false |
 *   | log | true |
 *   | force | false |
 *   | metaKey | false |
 *   | multiple | false |
 *   | scrollBehavior | top |
 *   | shiftKey | false |
 *   | timeout | 4000 |
 *   | waitForAnimations | true |
 * ```
 *
 * @remarks
 *
 * A preceding step like {@link When_I_find_element_by_text | "When I find element by text"} is required. For example:
 *
 * ```gherkin
 * When I find element by text "Text"
 *     And I click 80px and 75px
 * ```
 *
 * @see
 *
 * - {@link When_I_click | When I click}
 * - {@link When_I_click_position | When I click position}
 */
function When_I_click_x_y_coordinates(x, y, options) {
    (0, utils_1.getCypressElement)().click(x, y, (0, utils_1.getOptions)(options));
}
exports.When_I_click_x_y_coordinates = When_I_click_x_y_coordinates;
(0, cypress_cucumber_preprocessor_1.When)('I click {int}px and {int}px', When_I_click_x_y_coordinates);
/**
 * When I click on button:
 *
 * ```gherkin
 * When I click on button {string}
 * ```
 *
 * If multiple buttons are found, it will click on the first one.
 *
 * @example
 *
 * ```gherkin
 * When I click on button "Button"
 * ```
 *
 * With [options](https://docs.cypress.io/api/commands/get#Arguments):
 *
 * ```gherkin
 * When I click on button "Button"
 *   | log | true |
 *   | timeout | 4000 |
 *   | withinSubject | null |
 *   | includeShadowDom | false |
 * ```
 *
 * @see
 *
 * - {@link When_I_click_on_text | When I click on text}
 */
function When_I_click_on_button(text, options) {
    (0, queries_1.When_I_find_button_by_text)(text, options);
    (0, utils_1.getCypressElement)().click();
}
exports.When_I_click_on_button = When_I_click_on_button;
(0, cypress_cucumber_preprocessor_1.When)('I click on button {string}', When_I_click_on_button);
/**
 * When I click on link:
 *
 * ```gherkin
 * When I click on link {string}
 * ```
 *
 * If multiple links are found, it will click on the first one.
 *
 * @example
 *
 * ```gherkin
 * When I click on link "Link"
 * ```
 *
 * With [options](https://docs.cypress.io/api/commands/contains#Arguments):
 *
 * ```gherkin
 * When I click on link "Link"
 *   | matchCase | true |
 *   | log | true |
 *   | timeout | 4000 |
 *   | includeShadowDom | false |
 * ```
 *
 * @see
 *
 * - {@link When_I_click_on_text | When I click on text}
 */
function When_I_click_on_link(text, options) {
    cy.contains('a', text, (0, utils_1.getOptions)(options)).first().click();
}
exports.When_I_click_on_link = When_I_click_on_link;
(0, cypress_cucumber_preprocessor_1.When)('I click on link {string}', When_I_click_on_link);
/**
 * When I click on text:
 *
 * ```gherkin
 * When I click on text {string}
 * ```
 *
 * Alternative:
 *
 * - {@link When_I_click | When I click}
 *
 * @example
 *
 * ```gherkin
 * When I click on text "Text"
 * ```
 *
 * With [options](https://docs.cypress.io/api/commands/contains#Arguments):
 *
 * ```gherkin
 * When I click on text "Text"
 *   | matchCase | true |
 *   | log | true |
 *   | timeout | 4000 |
 *   | includeShadowDom | false |
 * ```
 *
 * @see
 *
 * - {@link When_I_click_on_button | When I click on button}
 * - {@link When_I_click_on_label | When I click on label}
 * - {@link When_I_click_on_link | When I click on link}
 */
function When_I_click_on_text(text, options) {
    cy.contains(text, (0, utils_1.getOptions)(options)).click();
}
exports.When_I_click_on_text = When_I_click_on_text;
(0, cypress_cucumber_preprocessor_1.When)('I click on text {string}', When_I_click_on_text);
/**
 * When I click on label:
 *
 * ```gherkin
 * When I click on label {string}
 * ```
 *
 * @example
 *
 * ```gherkin
 * When I click on label "Label"
 * ```
 *
 * With [options](https://docs.cypress.io/api/commands/get#Arguments):
 *
 * ```gherkin
 * When I click on label "Label"
 *   | log | true |
 *   | timeout | 4000 |
 *   | withinSubject | null |
 *   | includeShadowDom | false |
 * ```
 *
 * @see
 *
 * - {@link When_I_click_on_text | When I click on text}
 */
function When_I_click_on_label(text, options) {
    (0, queries_1.When_I_find_element_by_label_text)(text, options);
    (0, utils_1.getCypressElement)().click();
}
exports.When_I_click_on_label = When_I_click_on_label;
(0, cypress_cucumber_preprocessor_1.When)('I click on label {string}', When_I_click_on_label);
